ingest/src/GitHubDataSource.ts

```diff
@@ -47,53 +48,44 @@ export const makeGitHubDataSource = async ({
   repoLoaderOptions,
   handleDocumentInRepo,
 }: MakeGitHubDataSourceArgs): Promise<DataSource> => {
-  const loader = new GithubRepoLoader(repoUrl, repoLoaderOptions);
-  return {
+  const repoOptions: Record<string, string | number> = {};
+  const { branch, recursive } = repoLoaderOptions ?? {};
+  if (branch) {
+    repoOptions["--branch"] = branch;
+  }
+  if (recursive) {
+    repoOptions["--recursive"] = `${recursive}`;
+  }
+
+  // Langchain GitHubRepoLoader consumes a lot of API requests and quickly hits
+  // the limit. This is now using makeGitDataSource as a drop-in replacement.
```

Comment:
Do we need the context comment?

Reply:
Prob not!

Reply:
Actually, I think I will leave it because it really does seem weird that we have this source with a weird API just to wrap another source